pkg/monitoring/rules/rules.go

```diff
@@ -0,0 +1,49 @@
+package rules
+
+import (
+	"github.com/machadovilaca/operator-observability/pkg/operatorrules"
+	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/pkg/monitoring/rules/alerts"
+	"kubevirt.io/containerized-data-importer/pkg/monitoring/rules/recordingrules"
+)
+
+const (
+	ruleName = "prometheus-cdi-rules"
+)
+
+// SetupRules initializes recording and alert rules in a namespace.
+func SetupRules(namespace string) error {
+	if err := recordingrules.Register(namespace); err != nil {
+		return err
+	}
+
+	if err := alerts.Register(namespace); err != nil {
+		return err
+	}
```

Comment:
I think you want these (or the Register func content itself) wrapped up in a https://pkg.go.dev/sync#Once
Since there is basically no guarantee that the caller won't run SetupRules multiple times

How does the k8s metrics helper lib achieve this?

Reply:
we do it the same for all operators at the moment, we have to set it up in order to list it...
@machadovilaca WDYT?

Reply:
end result is the same as using `prometheus.Register` directly, we tried to avoid different usage logic, it will error on duplicate registration

Reply:
I mean this Register(), which just appends
https://github.com/kubevirt/containerized-data-importer/pull/3068/files#diff-1fd933a1e8d6a957bbdb03b4e22f73c2b114766e15a46162de2dc20a90725f2eR6

@avlitman initially had the call to SetupMetrics inside the reconcile loop, which overwhelmed the PrometheusRule resource, which we probably want to protect against

Reply:
ah yes, should be idempotent in the dependency

Reply:
fixed in the lib.

Reply:
I sent on private, here again: https://github.com/machadovilaca/operator-observability/commits/main/